mesonbuild/mdevenv.py

```diff
@@ -28,54 +28,14 @@ def get_windows_shell() -> str:
     result = subprocess.check_output(command)
     return result.decode().strip()
 
-def get_wine_shortpath(build_dir: str, winecmd: str, wine_paths: T.List[str]) -> T.List[str]:
-    '''
-    WINEPATH size is limited to 1024 bytes which can easily be exceeded when
-    adding the path to every dll inside build directory. See
-    https://bugs.winehq.org/show_bug.cgi?id=45810.
-
-    To shorthen it as much as possible we use path relative to builddir
-    where possible and convert absolute paths to Windows shortpath (e.g.
-    "/usr/x86_64-w64-mingw32/lib" to "Z:\\usr\\X86_~FWL\\lib").
-    '''
-    rel_paths = []
-    abs_paths = []
-    builddir = Path(build_dir)
-    for p in wine_paths:
-        try:
-            rel = Path(p).relative_to(builddir)
-            rel_paths.append(str(rel))
-        except ValueError:
-            abs_paths.append(p)
-    if not abs_paths:
-        return rel_paths
```

Comment:
Is this no longer needed? It doesn't seem to be present in the consolidated version anymore.

Reply:
It was using paths relative to top builddir for devenv which has the annoying effect that you can only run wine from that directory. I was fine with that limitation because I thought we had no other choice, but now that we found that newer wine got fixed, I think it's better to not keep such fragile workaround. Also, we cannot rely on that hack for other use-cases because they can have a different workdir.

Reply:
I was actually thinking it would work better in the non-devenv case since there's a set workdir. e.g. maybe it could take a `cwd=None` kwarg and make it relative to that?

Reply:
Good point, did that now. But I noticed we set `Z:` prefix on all paths. That's probably going to break the relative path detection. Not sure why we have that, seems to work fine without `Z:` in devenv.

Reply:
Yes, I think it's better to remove that, it creates issues that users won't understand.

Reply:
Removed from devenv now.